Cleanup tron scribereader

yaroliakh opened the pull request:
Scribereader is not used anymore in Tron. Instead S3LogsReader will be used by default. 
Check commits for easier review

yaroliakh commented on tests/utils/scribereader_test.py:
```diff
@@ -127,300 +125,39 @@ def test_read_log_stream_for_action_run_yelp_clog_tz(local_datetime, expected_da
-        autospec=True,
-    ), mock.patch(
-        "staticconf.read", autospec=True, side_effect=static_conf_patch({"logging.max_lines_to_display": 1000})
-    ), mock.patch(
-        "tron.config.static_config.load_yaml_file",
-        autospec=True,
-    ):
-        # all the data we want is from the past, so we should only check the reader
-        mock_stream_reader.return_value.__enter__.return_value = iter(
-            [
-                """{
-                "tron_run_number": 1234,
-                "component": "stdout",
-                "message": "line 0",
-                "timestamp": "2021-01-02T18:10:09.169421619Z",
-                "cluster": "fake"
-            }""",
-            ]
-        )
-        # so lets make sure we don't call the tailer
-        mock_stream_tailer.return_value.__iter__.side_effect = Exception
-        output = read_log_stream_for_action_run(
-            action_run_id="namespace.job.1234.action",
-            component="stdout",
-            min_date=min_date,
-            max_date=max_date,
-            paasta_cluster="fake",
-        )
-
-    mock_stream_reader.assert_called_once_with(
-        stream_name="stream_paasta_app_output_namespace_job__action",
-        min_date=min_date,
-        max_date=max_date,
-        reader_host="host",
-        reader_port=1234,
-    )
-    mock_stream_tailer.assert_not_called()
-    assert output == ["line 0"]
-
-
```
those are scribereader specific tests that don't apply to S3LogsReader (there is no difference in whether start and end dates are the same or not)

nemacysts reviewed (dismissed):
thanks for cleaning up!

yaroliakh commented on yelp_package/extra_requirements_yelp.txt:
```diff
@@ -1,14 +1,4 @@
 clusterman-metrics==2.2.1  # used by tron for pre-scaling for Spark runs
-dateglob==1.1.1 # required by yelp-logging
-geogrid==2.1.0  # required by yelp-logging
-monk==3.0.4     # required by yelp-clog
-ply==3.11   # required by thriftpy2
-scribereader==1.1.1  # used by tron to get tronjob logs
-simplejson==3.19.2  # required by yelp-logging
-srv-configs==1.3.4  # required by monk
-tenacity==8.2.3 # required by yelp-logging
-thriftpy2==0.4.20   # required by monk
-yelp-cgeom==1.3.1   # required by geogrid
-yelp-clog==7.1.4  # scribereader dependency
-yelp-logging==4.17.0  # scribereader dependency
```
ah, actually `yelp-clog` still pulls a bunch of dependencies that aren't actually used. I'll see if it's possible to make them optional

yaroliakh commented:
updated a bit the PR to completely replace clog with logreader package. Removed any `clog` references and updated the requirements list (it shouldn't be dragging any extra deps now)
